src/IceRpc.Slice.Generators/Internal/Emitter.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) ZeroC, Inc.
+
+namespace IceRpc.Slice.Generators.Internal;
+
+internal class Emitter
+{
+    internal string Emit(IReadOnlyList<ServiceClass> serviceClasses, CancellationToken cancellationToken)
+    {
+        var generatedClasses = new List<string>();
+        foreach (ServiceClass serviceClass in serviceClasses)
+        {
+            // stop if we're asked to.
+            cancellationToken.ThrowIfCancellationRequested();
+
+            string methodModifier =
+                serviceClass.HasBaseServiceClass ? "public override" :
+                serviceClass.IsSealed ? "public" : "public virtual";
+
+            string dispatcherBlock;
```

Comment:
We should not have a variable called "dispatcherBlock" and another called "dispatchBlocks". That's too confusing. Also, what is a "block"?

Reply:
it was to mean a code block

Reply:
Reworked this code